### PR TITLE
Add Trino 394 release notes

### LINKED ISSUE #13723: Release notes for 394

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 394 release.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Docs only.

> How would you describe this change to a non-technical end user or system administrator?

Add the release notes for Trino 394 with sections for general, security, connectors and other topics as included in the release.

## Related issues, pull requests, and links

Fixes #13723

## Documentation

( ) No documentation is needed.
(✅) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(✅) No release notes entries required.
( ) Release notes entries required with the following suggested text:

## Verification for each commit

Format: commit, committer/maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 17 August 2022 ✅ 

59618ecb405a8379785c7c5fd82519529427e06e @ebyhr #13659 ✅ rn ✅ docs
6602cb92b7c571b95113b2d652a2a25af6bd241f @ebyhr #13720 ✅ rn ✅ docs
dbe6e2bd1367c739d2ad2b0dcb38df0552cccd5e @ebyhr #13716 ✅ rn ✅ docs

## 18 August 2022 ✅ 

41d974986356c2d52b4adaa136e582e9d515fd8c @ebyhr #13706 ✅ rn ✅ docs
98fd1ade66a0c0ac6a2e4bb5cd1f02568610430a @martint #13735 ✅ rn ✅ docs
e1dc3e62f6c5fe1ebb15ed6bf11b9bce534f73c1 @electrum #13719 ✅ rn ✅ docs
ee57bc40fbdedaa783f18b0ee8cf1ffde92be272 @martint #13734 ✅ rn ✅ docs
53d9e11d0452e80c59f75b920d3effce269f8a77 @martint  #13734 ✅ rn ✅ docs
c48d1b19549e0306c22e90db4fc5abd12ef47d9c @arhimondr #13717 ✅ rn ✅ docs
adcfbe95250f216170046d1ab8fb108b61d1c61e @ebyhr #13614 ✅ rn ✅ docs
55fd93421d02e7e3a92103baf08eda2d09c84a0f @raunaqmorarka #13732 ✅ rn ✅ docs
d23746c4eafac17aad5f7ac8df1ed2e1449c4bdc @raunaqmorarka #13732 ✅ rn ✅ docs
b4b7f5fb8c1227263d3ec0db2cd0d0d8d3a9d605 @Praveen2112 #13700 ✅ rn ✅ docs
7c2f0c21bf987501481bbfc364e0a661346a195e @Praveen2112 #13700 ✅ rn ✅ docs
edabca20d1c3a0883513f1f860af636a7f4d7740 @electrum #13736 ✅ rn ✅ docs
257c2a5ac47a731651b43cc7af35f83bf5a034c4 @electrum #13736 ✅ rn ✅ docs

## 19 August 2022 ✅ 

ab7a4de47e39cced27799cb1f361d2ac525467de @raunaqmorarka #13432 ✅ rn ✅ docs
a59643b5fef0b99d6e0f2b3458e05ad6bb05be85 @raunaqmorarka #13432 ✅ rn ✅ docs
1b6724a045a883628b42e8711f2e530be7731093 @raunaqmorarka #13432 ✅ rn ✅ docs
0998cf859c5a38423cdcd60778e6db7500fdc120 @raunaqmorarka #13554 ✅ rn ✅ docs
8494caa5ffc67a1281a31dfd7426e0a71fe8bf0d @martint #10873 ✅ rn ✅ docs
6676c2818b8881a87ccc6911111dd64f8786150c @raunaqmorarka #13549 ✅ rn ✅ docs
2be6e4495f017a7612fc9167c2b062ee04e7fb89 @martint #13644 ✅ rn ✅ docs
5938f18d19799ac6b4fc5668266465c4054bfc05 @ebyhr #12502 ✅ rn ✅ docs
6b24b04d80e7817ba73178063123946b61a2e3ee @ebyhr #12502 ✅ rn ✅ docs
a52a964f466a452badd3275ac840b02267e891f5 @ebyhr #12502 ✅ rn ✅ docs

## 20 August 2022 ✅ 

2b1423976c1c1055ebf1a4bd26514f126dfa6577 @kokosing #13731 ✅ rn ✅ docs
111b3eed0ca4f97ce305702a26825c216b4c3fb5 @kokosing #13731 ✅ rn ✅ docs
93c614b6dd8c71273e4c4ff04a2af78c29554b41 @ebyhr #13763 ✅ rn ✅ docs

## 21 August 2022 ✅ 

66c196ac6e5ed6212391be300840bcf7f41f855b @ebyhr #13429 ✅ rn ✅ docs
8930c21b604a2ccb1aee88f3841472d99168373d @ebyhr #13429 ✅ rn ✅ docs
852a41e9f42b9f59aff3e8216edfc1f3c49edc63 @ebyhr #13429 ✅ rn ✅ docs

## 22 August 2022 ✅ 

98e5607fdb08005979bc9cefcd857339398970e1 @hashhar #13486 ✅ rn ✅ docs
32740f103783435faf8a14c475e8f5e95d6abd3f @ebyhr #13767 ✅ rn ✅ docs
280248de62648ec2e657d7560bd66373b7c61c59 @findepi #13647 ✅ rn ✅ docs
21441811281ab950a799f9581b4cb9bd0a67277d @findepi #13645 ✅ rn ✅ docs
ae736a51d0e7f1532035242910a23202e9bbb4a8 @findepi #13646 ✅ rn ✅ docs
7c92a16f0083183283f8f68ce9eaf0e9d1209e8f @arhimondr #13025 ✅ rn ✅ docs
792156d281bb010a0397528349359d6efee6af44 @arhimondr #13025 ✅ rn ✅ docs
350dcbf8d61e41d59fbfbcfd99ecd70cf94c8c9b @arhimondr #13025 ✅ rn ✅ docs
57f8179626d8edb7dc72ca218fd9a0a36666b61d @arhimondr #13025 ✅ rn ✅ docs
aea3acc1677afafbda92af6df416ea4ac99d5ab2 @arhimondr #13025 ✅ rn ✅ docs
a1e98b512c58c4c99e42e60a949a1ca36800f4f3 @arhimondr #13025 ✅ rn ✅ docs
89dbd1183ef231580864bd7ef8bcea7af4529e1a @arhimondr #13025 ✅ rn ✅ docs
af2e92fb09fb00a74cfde2a6c9b9506a991cf11e @arhimondr #13025 ✅ rn ✅ docs
01658e04c7e93678df47a8eb127ef8f6c7599ecf @arhimondr #13025 ✅ rn ✅ docs
6197144732f344dd9602ee633b9b722b57715f4f @arhimondr #13025 ✅ rn ✅ docs
e64998e3c04ba434926bda4ec115e2b972db67ab @arhimondr #13721 ✅ rn ✅ docs
c2394b2f016af3cf2c4e5c4b4e02735a2b6671ab @kokosing #13748 ✅ rn ✅ docs
 f8bd4a0b5e8fa59eae0b588e613042221cf703e8 @sopel39 #13765 ✅ rn ✅ docs
ee818a3ce3e1c5a02627e77a968819e49267e442 @sopel39 #13765 ✅ rn ✅ docs
21b1adaf0d0940a01f9e1416f2f8b3c513bfedee @sopel39 #13765 ✅ rn ✅ docs
5374ba7062e4b74a9119882a4bc8715fff32cca1 @martint #13758 ✅ rn ✅ docs
18f17b2b91ba6fb97d017ea9d2f9c4dbad0b4e55 @martint #13021 ✅ rn ✅ docs
b09658ce6c2dd89de6051fb61974eb73d2916ac4 @martint #13021 ✅ rn ✅ docs
2f62586c12cef3c3a634dd9a5ab59c459d5f06b4 @martint #12976 ✅ rn ✅ docs
e7c46c1760cd88efd6842f03b289d1280f86498d @martint #13744 ✅ rn ✅ docs
edb13ac27b5988d4fecdfe43d116aad44bd0d3fc @martint #13744 ✅ rn ✅ docs
739b5c82037dd50e7d3ef82d149d170c0e5a25f8 @ebyhr #13743 ✅ rn ✅ docs
9b7a7bfb2c4969ca4e13d7e56fcb994130d4e7f5 @ebyhr #13743 ✅ rn ✅ docs
90d3677ca09fe2d792048e6aa98f83d27a430622 @ebyhr #13760 ✅ rn ✅ docs
ca6a3013ea36727a2befe8726a96ca35b5c361ee @ebyhr #12800 ✅ rn ✅ docs
a76fd0b5d6095d2109d10f3541b408a81c479437 @ebyhr #12800  ✅ rn ✅ docs
52bf6680c1b25516f6e8e64f82ada089abc0c9d3 @ebyhr #12800 ✅ rn ✅ docs
9bc036eceea3bd325541c46b9ebffcc90532ee56 @ebyhr #13094 ✅ rn ✅ docs
24b45481a89497da934a09ea27dd6390bc97e5fd @ebyhr #13094 ✅ rn ✅ docs
c555a1325fda0458119bab27730948e41770eee7 @ebyhr #13094 ✅ rn ✅ docs
80d6b6cac6b5ff89e8d1ef72bcf8923e274cc545 @ebyhr #13094 ✅ rn ✅ docs

## 23 August 2022 ✅ 

4b0408df9012e4912c6c0de04dc738a925248cdb @findepi #13755 ✅ rn ✅ docs
daa858da3027cdc153f74e6cae30d1e04615957c @findepi #13774 ✅ rn ✅ docs
d4e97dacb17ab763fa701d77d561cdce4c8bb378 @ebyhr #13746 ✅ rn ✅ docs
12b0c3dc1c3aad717b06669caf73cadd84ec409d @ebyhr #13785 ✅ rn ✅ docs
b0f4f73eda0131e3a5c5bd25ff89f5ad7758483f @findepi #13561 ✅ rn ✅ docs
9888b846e5012dcd65a4c74b019ab28b311d8be4 @findepi #13561 ✅ rn ✅ docs
af4bae2bd6f4db6c89839bc9372eb36b4533803f @electrum #13696 ✅ rn ✅ docs
d16d177e157a6ef21113d4d47622e70a0615504b @electrum #13696 ✅ rn ✅ docs
f8a4520ca3df02094a557667650ec8ce989ce35d @electrum #13696 ✅ rn ✅ docs
0ee59bd53b069b33ea0b1b4ded563609d89e12ef @electrum #13696 ✅ rn ✅ docs
14f61898fa50411af7943578b2651fc2d1d6a7ff @electrum #13696 ✅ rn ✅ docs
1759cb5e69c9ec3c864ef22e8c961063707de02e @electrum #13696 ✅ rn ✅ docs
f396cbed8460382356c37b077a2e21df3e8b59df @findepi #13793 ✅ rn ✅ docs
9f971eefbc188ffc0dfd524759bf0ee07d8b4188 @findepi #13793 ✅ rn ✅ docs
a6a0b6b172e255cfb9ed1254e726a61a14e20d25 @findepi #13796 ✅ rn ✅ docs
624fedb3cf139f76f9f2e1d773773f75afa01fe1 @arhimondr #13799 ✅ rn ✅ docs

## 24 August 2022

d4288f6fffca2bf850bd1c8c50ff2f6416694b29 @hashhar #13812 ✅ rn ✅ docs
42f3b8a3bf158352b11ec34dc14be8dbd5ea5ad4 @hashhar #13812 ✅ rn ✅ docs
8fea27c974e5078e8d1105fdb56a2dd417f7750b @findepi #13811 ✅ rn ✅ docs
1e61a7ce05ce21f9f54f18cf8ba8e38187e38d67 @hashhar #13816 ✅ rn ✅ docs
9ce301813e2de8bc3137e2664fb6b54112f34c3f @ebyhr #13820 ✅ rn ✅ docs
19c2bc45aaa4e5eb90c33c910628cbf09da2b093 @arhimondr #13421 ✅ rn ✅ docs
d9402bc77994d858672313b68cacac623d8f3c14 @arhimondr #13421 ✅ rn ✅ docs
170faf53cd1c6a1a39118ae6d9475f21926b3885 @arhimondr #13421 ✅ rn ✅ docs
a183959a962b8c9b012547afc02d566a10cf2591 @arhimondr #13421 ✅ rn ✅ docs
43d61ab04bef3cbc844193d1f8cc616778fc1cc1 @arhimondr #13421 ✅ rn ✅ docs
c9e27f895fb78fab06b9aeb422ed56cb37b79858 @arhimondr #13421 ✅ rn ✅ docs
26ba659305d698eafb5b1d6f5a98a49f27d480f8 @arhimondr #13421 ✅ rn ✅ docs
5f685f5cc0fb27fc1cf50dfe2e4b6c2b479b0833 @arhimondr #13421 ✅ rn ✅ docs
7990d40ea28f548aefde10d1eed147dfe741b84f @arhimondr #13421 ✅ rn ✅ docs
d0b0994531c8ebd2c38a88592400410f1dc31d17 @arhimondr #13421 ✅ rn ✅ docs
a977d88b051e6616d1fd9c355362627383abec29 @losipiuk #13798 ✅ rn ✅ docs
f7ad25addca4aad78acda38bc947523c3a7f88d6 @ebyhr #13742 ✅ rn ✅ docs
0547b321937befa4ed95ea2b065dd16ec4bbecf9 @ebyhr #13742 ✅ rn ✅ docs
d9aa5b328345218916deba29aad5274d98ce577a @ebyhr #13742 ✅ rn ✅ docs
fe16cae089054aa61d1f443c01b4cdc01fd55443 @ebyhr #13742 ✅ rn ✅ docs
d393e3689f847b720ececef3b808d2725ce4da98 @martint #12530 ✅ rn ❌ docs
5a6fefb3fa9dff439c49c4628e3a8627ee2612b1 @martint #13479 ✅ rn ✅ docs
c17a5522162d99784936698c3209c00c5b3cac2c @ebyhr #13842 ✅ rn ✅ docs
e051a0f2f8385dea9efbab3c866e23403f895e01 @ebyhr #13840 ✅ rn ✅ docs

## 25 August 2022

05110214887b33d07925e306be970ab8c8fa2843 @findepi #13559 ❌ rn ✅ docs
cdbb5f872a242b7fb8cf6500a7ed2cdeae9a999a @findepi #13829 ✅ rn ✅ docs
b4f77ceda7aab5e1a6ec6ec60fec51a908af8537 @findepi #13829 ✅ rn ✅ docs
138eaaad0b0b97f2cecb10bd097a53cf3eebc260 @findepi #12153 ✅ rn ✅ docs
4528e91438bbeeb5b9f5d6788f41c3174cd9303c @findepi #13823 ✅ rn ✅ docs
7f0f6fac4c6eccd7cb15cf71ba48ea4abbb31746 @findepi #13823 ✅ rn ✅ docs
83fbbb9657d261be9587afc16347e5a3083d6d44 @findepi #13823 ✅ rn ✅ docs
3b76ce44032ad6f42a084de3c8e6d61848d15f26 @findepi #13823 ✅ rn ✅ docs
b487c74d004883c08664b69ba776897e286275cb @findepi #13823 ✅ rn ✅ docs
1a7c59f70a8e23d1f8b173e343bf0f002aad3a6f @findepi #13819 ✅ rn ✅ docs
5574a4e6d4edb5bbb4e8f6c5f437f519147193f9 @ebyhr #13828 ✅ rn ✅ docs
e59e948c75f809a17df5bfccfa99cdd2e782dc43 @sopel39 #13845 ✅ rn ✅ docs
473e2fd5a5c106adf53030b1c1c34711b7b8a8c9 @sopel39 #13453 ✅ rn ✅ docs
95d2994b0a3cc7c4d9cfa5f8a069d20836e18b2d @findepi #13824 ✅ rn ✅ docs
875727157a71082c97d8b579e6022b86aa4afc4e @findepi #13824 ✅ rn ✅ docs
46e226e077521cddd35cf446ab9b6b09304289e7 @findepi #13831 ✅ rn ✅ docs
20fa89a2d5e406635c012028b0f94ff057a98e67 @ebyhr #13856 ✅ rn ✅ docs
9719f9bd248751ce754d8636efbcc0b48f470cd6 @ebyhr #13856 ✅ rn ✅ docs

## 26 August 2022 ✅ 

07aefd93678bb6469ed9bd798f9d1d74ee474e67 @hashhar #13853 ✅ rn ✅ docs
349f81bf61af4286911a72a18f90c8e316dd00ac @losipiuk #13818 ✅ rn ✅ docs
e98fb9018fe33be9a5239379312635e448e165e3 @losipiuk #13818 ✅ rn ✅ docs
a8f75d3cb83ba14c1ab49cc86b7a449dee0415a5 @sopel39 #13757 ✅ rn ✅ docs
303b80554dedba351757780930d0bc460514979e @sopel39 #12968 ✅ rn ✅ docs
f674632452930fa10393de8a160b1ac3162b470d @sopel39 #12968 ✅ rn ✅ docs
06ce5b99b49b2a6cc1f860c0ad637058b807ed0a @sopel39 #12968 ✅ rn ✅ docs
240ed0a0ef193d509248026dd66ce3bbd7b5178c @sopel39 #12968 ✅ rn ✅ docs
7f3047691383beb5320900b0adb94e684dbfe82a @sopel39 #12968 ✅ rn ✅ docs
114435c542f2c641dba0f100a4f2bc0203f615ae @sopel39 #12968 ✅ rn ✅ docs
d0563dc41d0ea6f1cbb2e59b2d44cd32e5c2430e @sopel39 #12968 ✅ rn ✅ docs
1c121edeae29c323d3920f870fba2ed933d79c6c @losipiuk #13859 ✅ rn ✅ docs
26d4fcb0c7b71e31f14e24e253507b541470d398 @ebyhr #13861 ✅ rn ✅ docs
3647a9a8e2c538bf73698256f1d88b1d28edb18b @martint #13869 ✅ rn ✅ docs
56c97c56a9978f62ba726fff1efeb3512291f479 @martint #13410 ✅ rn ✅ docs

## 28 August 2022

01d2bfb0bf8de967ecbbafd3945646760c0ccb4b @Praveen2112 #13482 ✅ rn ❌ docs
5b9d8749fb3a6a6ef38a432c36dec1497120e2d9 @Praveen2112 #13482 ✅ rn ❌ docs
5120f3fe019f8e6ee2f94d6e954c7bf11ad63f02 @Praveen2112 #13482 ✅ rn ❌ docs

## 29 August 2022

fcef472551d103da978d55b23a1cafc82b01930c @ebyhr #13884 ✅ rn ✅ docs
8c20f4ea5dd943ecd19223791494dbac798e1c51 @findepi #13567 ✅ rn ✅ docs
135338815a8306d310dace5f451dbd4e6bc62e84 @findepi #13567 ✅ rn ✅ docs
cbb5eaa8455cf13de92315503e465f647dba272a @kasiafi #13893 ✅ rn ✅ docs
0fce1fcc280a1d066b8581b8804daff74c1bf023 @findepi #13887 ✅ rn ✅ docs

### CODE REVIEW

**Comment by colebow** [comment]
Sorry for the delay in getting this up, got back from a trip and came down with COVID. Will do my best to address feedback ASAP.

File: docs/src/main/sphinx/release/release-394.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Release 394 (24 August 2022)
+
+## General
+
+* Improve performance of joins which use a single `bigint` column. ({issue}`13432`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:5]: "Use a" or "join on" a single `bigint` column?

File: docs/src/main/sphinx/release/release-394.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+* Ensure `UPDATE` queries cannot insert null values into columns which are
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:6]: > ... into columns with a `NOT NULL` constraint

File: docs/src/main/sphinx/release/release-394.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  `NOT NULL`. ({issue}`13435`)
+* Fix failure when an `UPDATE` query uses a `WHERE` clause which always
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:8]: > ... when an `UPDATE` query contains a `WHERE` clause ...

File: docs/src/main/sphinx/release/release-394.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  evaluates to false. ({issue}`12422`)
+* Fix potential failure for queries involving joins and implicit or explicit
+  casts of `null` to a concrete type. ({issue}`13565`)
+
+## Docker image
+
+* Terminate the JVM if it is unable to create a native thread. ({issue}`13736`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:15]: This needs to be rephrased in a more user-friendly way. Users don't know or care about when a native thread can't be created.

File: docs/src/main/sphinx/release/release-394.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+## BigQuery connector
+
+* Add `query` table function for full query pass-through to the connector. ({issue}`12502`)
+* Add support for the `CREATE TABLE AS SELECT` and `INSERT` statements. ({issue}`6868`, {issue}`6869`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:20]: Split into two entries. They are separate changes with different associated PRs:

> * Add support for `CREATE TABLE ... AS SELECT ...`. ({issue}`6869`)
> * Add support for `INSERT`. ({issue}`6868`)

File: docs/src/main/sphinx/release/release-394.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+## Delta Lake connector
+
+* Disallow adding a `NOT NULL` column to a table which is not empty. ({issue}`13785`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:24]: > ... adding a column with a `NOT NULL` constraint to a table that is not empty.

File: docs/src/main/sphinx/release/release-394.md (hunk 4 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-394.md:15]: @electrum any recommendations?

File: docs/src/main/sphinx/release/release-394.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
+# Release 394 (31 August 2022)
+
+## General
+
+* Add support for `EXPLAIN (TYPE DISTRIBUTED, FORMAT JSON)`. ({issue}`12968`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:5]: I would describe this in prose to that it's clear what's new:

* Add support for JSON format to output of `EXPLAIN`

File: docs/src/main/sphinx/release/release-394.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+* Improve performance of joins which join on a single `bigint` column. ({issue}`13432`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:6]: > Improve performance of queries involving joins on a single `bigint` column.

File: docs/src/main/sphinx/release/release-394.md [hunk 9]
```diff
@@ -26,16 +28,61 @@
   empty. ({issue}`13785`)
 * Fix failure when reading Parquet data that contains only null values. ({issue}`9424`)
 
+## Druid connector
+
+* Improve query performance when a `varchar` column is interpreted as containing
+  date values in the `uuuu-MM-dd` format and is filtered on. ({issue}`12925`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:34]: What's `uuuu`? Why the difference in case for `uuuu` and `dd` vs `MM`?

Also, the grammar for this sentence is off. I would rephrase as:
> Improve performance of queries that perform filtering on varchar columns that contain temporal data with the format `YYYY-MM-DD`

File: docs/src/main/sphinx/release/release-394.md [hunk 10]
```diff
@@ -26,16 +28,61 @@
...
+
+## Elasticsearch connector
+
+* Add support for multiple hosts in the `elasticsearch.host` configuration
+  property. ({issue}`12530`)
+
 ## Hive connector
 
 * Fix failure when reading Parquet data that contains only null values. ({issue}`9424`)
+* Fix failure when reading duplicated column statistics. ({issue}`13787`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:44]: > Fix failure when the metastore returns duplicated column statistics.

File: docs/src/main/sphinx/release/release-394.md [hunk 11]
```diff
@@ -26,16 +28,61 @@
...
 
 ## Iceberg connector
 
+* Improve query planning performance when a `varchar` partitioning column
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:48]: YYYY-MM-DD ?

File: docs/src/main/sphinx/release/release-394.md [hunk 12]
```diff
@@ -26,16 +28,61 @@
...
+  date values in the `uuuu-MM-dd` format and is filtered on. ({issue}`12925`)
+
+## Phoenix connector
+
+* Improve query performance when a `varchar` column is interpreted as containing
+  date values in the `uuuu-MM-dd` format and is filtered on. ({issue}`12925`)
+
 ## Pinot connector
 
+* Add support for TLS connection when the `pinot.controller-urls` configuration
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:66]: I would simplify to:

> Add support for TLS when connecting to the Pinot controllers and brokers.

File: docs/src/main/sphinx/release/release-394.md (hunk 9 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-394.md:34]: I think there's some semantic difference between YYYY and uuuu because uuuu goes negative for dates before the year 0 and handles epoch start time differently? @findepi can you clarify if this absolutely needs to be `uuuu`, because otherwise Martin's proposal would be accurate and more user-friendly.

File: docs/src/main/sphinx/release/release-394.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
+# Release 394 (29 August 2022)
+
+## General
+
+* Add support for JSON format as output of `EXPLAIN`. ({issue}`12968`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-394.md:5]: maybe

Add JSON as output format of EXPLAIN

File: docs/src/main/sphinx/release/release-394.md (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-394.md:17]: Maybe 

* Ensure Trino stops running with insufficient resources to avoid partial failures.

File: docs/src/main/sphinx/release/release-394.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+* Add support for TLS when connecting to the Pinot controllers and brokers. ({issue}`13410`)
+* Fix query failure when using the `HAVING` clause. ({issue}`13429`)
+
+## PostgreSQL connector
+
+* Improve query performance when a `varchar` column is interpreted as containing
+  date values in the `YYYY-MM-DD` format, is filtered on, and the
+  `postgresql.experimental.enable-string-pushdown-with-collate` configuration 
+  property is enabled. ({issue}`12925`)
+* Prevent using a column name which is longer than the max length supported by
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-394.md:75]: maximum length

File: docs/src/main/sphinx/release/release-394.md [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+  PostgreSQL. Previously, long names were truncated to the max length. ({issue}`13742`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-394.md:76]:
```suggestion
  PostgreSQL. Previously, long names were truncated. ({issue}`13742`)
```

File: docs/src/main/sphinx/release/release-394.md [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+
+## SQL Server connector
+
+* Prevent renaming a column to a name which is longer than the max length
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-394.md:80]: see earlier

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-394.md [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+  contain temporal data with the format `YYYY-MM-DD`. ({issue}`12925`)
+
+## Pinot connector
+
+* Add support for TLS when connecting to the Pinot controllers and brokers. ({issue}`13410`)
+* Fix query failure when using the `HAVING` clause. ({issue}`13429`)
+
+## PostgreSQL connector
+
+* Improve query performance when a `varchar` column is interpreted as containing
```
Comment (martint) [inline docs/src/main/sphinx/release/release-394.md:73]: The grammar of this sentence should be adjusted like the other instances.

File: docs/src/main/sphinx/release/release-394.md (hunk 9 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-394.md]: I don't think it matters. The point is to describe to users what the general shape is. No one is going to know what `uuuu` means (where is that defined?).

Comment (findepi) [reply docs/src/main/sphinx/release/release-394.md]: `YYYY-MM-DD` works from informal perspective.

**Comment by findepi** [comment]
> [0fce1fc](https://github.com/trinodb/trino/commit/0fce1fcc280a1d066b8581b8804daff74c1bf023) @findepi #13887 ✅ rn ✅ docs

That's a bad link. It's a link to an unrelated issue.
Is the commit list assembled by hand?
